src/anchor_in_depth/milestone_project_ido-program.md

```diff
@@ -0,0 +1,1320 @@
+# Project IDO Program
+
+## Overview
+
+It is now time to implement the in-depth concepts you learned in section 4 of the book through a project. Create a new anchor workspace
+
+```bash
+anchor init ido-program
+```
+
+As the name suggests, this is a program that helps projects do an Initial DEX Offering to investors.
+
+This program will be doing a couple of things. In an IDO, a project distributes its native tokens among willing investors. There are a few types of IDOs which are done. We will be implementing a fair launch IDO platform. In here, a project escrows the native tokens into a pool. When the IDO opens, investors come and deposit their fiat tokens into the pool. After the IDO is over, investors can claim their native tokens, the amount of which is calculated by the the pool percentage share of their deposit. The project can also finally withdraw all the deposited fiat tokens.
+
+We recommend keeping programs in a single `lib.rs` file.
```

Comment:
Why do we recommend this?

Reply:
Although maintaining a project directory structure is a good practice, implementing it in the milestone project might make the Anchor Book content massive.